pkg/argo/fetch.go

```
@@ -40,6 +43,32 @@ func newFetchResponse(newReq *pb.FetchRequest, eof bool, logs []string) *pb.Fetc
 	}
 }
 
+func getStepIdx(wf *v1alpha1.Workflow, targetStepGroup string) (int, error) {
+	stepIdx := 1
+	stepGroupName, e := getFirstStepGroup(wf, wf.ObjectMeta.Name)
+	if e != nil {
+		return -1, e
+	}
+	for {
+		if stepGroupName == targetStepGroup {
+			return stepIdx, nil
+		}
+		stepGroupName, e = getNextStepGroup(wf, stepGroupName)
+		stepIdx++
+		if e != nil {
+			return -1, e
+		}
+	}
+}
+
+func logViewURL(ns, wfID, stepID string) (string, error) {
+	ep := os.Getenv("SQLFLOW_ARGO_UI_ENDPOINT")
+	if ep == "" {
+		return "", fmt.Errorf("should set SQLFLOW_ARGO_UI_ENDPOINT if enable Argo mode")
```

Maybe raise this error in some early stage before the job have started running.